workers/tests/unittests/test_config.py

```diff
@@ -0,0 +1,102 @@
+import inspect
+import os
+import tempfile
+import unittest
+from unittest import mock
+
+from ohsome_quality_analyst.config import config
+
+
+class TestConfig(unittest.TestCase):
+    @mock.patch.dict("os.environ", {}, clear=True)
+    def test_get_config_path_empty_env(self):
+        self.assertEqual(
+            config.get_config_path(),
+            os.path.join(
+                os.path.dirname(
+                    inspect.getfile(config),
+                ),
+                "config.yaml",
+            ),
+        )
+
+    @mock.patch.dict(
+        "os.environ",
+        {"OQT_CONFIG": os.path.join(tempfile.gettempdir(), "config.yaml")},
+        clear=True,
+    )
+    def test_get_config_path_set_env(self):
+        self.assertEqual(
+            config.get_config_path(),
+            os.path.join(tempfile.gettempdir(), "config.yaml"),
+        )
+
+    def test_load_config_from_file(self):
+        path = config.get_config_path()
+        cfg = config.load_config_from_file(path)
+        self.assertIsInstance(cfg, dict)
+        self.assertTrue(cfg)
+
+    @mock.patch.dict("os.environ", {}, clear=True)
+    def test_load_config_from_env_empty(self):
+        cfg = config.load_config_from_env()
+        self.assertEqual(cfg, {})
+
+    @mock.patch.dict(
+        "os.environ",
+        {"OQT_GEOM_SIZE_LIMIT": "200", "POSTGRES_HOST": "foo"},
+        clear=True,
+    )
+    def test_load_config_from_env_set(self):
+        cfg = config.load_config_from_env()
+        self.assertDictEqual(
+            cfg,
+            {"geom_size_limit": "200", "postgres_host": "foo"},
+        )
+
+    @mock.patch.dict("os.environ", {}, clear=True)
+    def test_get_config(self):
+        cfg = config.get_config()
+        self.assertIsInstance(cfg, dict)
+        self.assertTrue(cfg)
+
+    def test_check_missing_conf_keys_compete(self):
+        cfg = {
+            "postgres_host": None,
+            "postgres_port": None,
+            "postgres_db": None,
+            "postgres_user": None,
+            "postgres_password": None,
+            "data_dir": None,
+            "geom_size_limit": None,
+            "log_level": None,
+            "ohsome_api": None,
+            "test_db": None,
+            "user_agent": None,
+            "datasets": None,
+        }
```

Comment:
I think `None` should be replaced through some values. See comment to `config.check_missing_conf_keys(…)`.

Reply:
Configuration setup has been changed to always use default values if no configuration has been provided